### DISCUSSION TITLE
Using ifcpatch as a library

### DISCUSSION THREAD

**Discussion by mehulshinde** [IfcOpenShell/IfcOpenShell#4366] [Q&A]
I used PyPI to install ifcopenshell in my MB M2 Max as the pre-build package kept giving me `IfcOpenShell not built for 'darwin/64bit/python3.10'` error.

I wanted to use ifcpatch for which I cloned the source and followed the steps as mentioner [here](https://blenderbim.org/docs-python/ifcpatch.html). I was able to use ifcpatch on CLI but have been unsuccessful importing and using it inside my python files. The doc doesn't mention any additional steps for using ifcpatch as a library.

TL;DR - how do I use ifcpatch as a library?

**Comment by Moult** [comment]
Something doesn't make sense here, if you can use IfcPatch on CLI, then you can use it as a library.

**Comment by mehulshinde** [comment]
I was having trouble importing the library outside the ifcpatch directory. The installation steps ask you to cd into the directory and that way it is possible to to import the library. To use it outside, it did the following:

```python
import sys
sys.path.append('IfcOpenshell/src/ifcpatch')
import ifcpatch
```
